xmtp_mls/src/client.rs

```diff
@@ -40,21 +42,24 @@ impl From<&str> for ClientError {
 }
 
 pub struct Client<ApiClient> {
-    pub api_client: ApiClient,
+    pub api_client: ApiClientWrapper<ApiClient>,
     pub(crate) _network: Network,
     pub(crate) _identity: Identity,
     pub store: EncryptedMessageStore, // Temporarily exposed outside crate for CLI client
 }
 
-impl<ApiClient> Client<ApiClient> {
+impl<ApiClient> Client<ApiClient>
+where
+    ApiClient: XmtpMlsClient + XmtpApiClient,
+{
     pub fn new(
         api_client: ApiClient,
         network: Network,
         identity: Identity,
         store: EncryptedMessageStore,
     ) -> Self {
         Self {
-            api_client,
+            api_client: ApiClientWrapper::new(api_client),
```

Comment:
I wrap the `ApiClient` into a prettier interface that abstracts away the Protobuf nonsense. Open to better names for  it than `ApiClientWrapper`.